crates/compiler/load_internal/src/file.rs

```diff
@@ -4127,234 +4163,7 @@ fn build_header<'a>(
-        for loc_exposed in exposes.iter() {
-            // Use get_or_insert here because the ident_ids may already
-            // created an IdentId for this, when it was imported exposed
-            // in a dependent module.
-            //
-            // For example, if module A has [B.{ foo }], then
-            // when we get here for B, `foo` will already have
-            // an IdentId. We must reuse that!
-            let ident_id = ident_ids.get_or_insert(loc_exposed.value.as_str());
-            let symbol = Symbol::new(home, ident_id);
-
-            exposed.push(symbol);
-        }
-
-        if cfg!(debug_assertions) {
-            home.register_debug_idents(ident_ids);
-        }
-
-        ident_ids.clone()
-    };
-
-    let package_entries = packages
-        .iter()
-        .map(|pkg| {
-            let pkg = pkg.value;
-            (pkg.shorthand, pkg.package_name.value)
-        })
-        .collect::<MutMap<_, _>>();
-
-    // Send the deps to the coordinator thread for processing,
-    // then continue on to parsing and canonicalizing defs.
-    //
-    // We always need to send these, even if deps is empty,
-    // because the coordinator thread needs to receive this message
-    // to decrement its "pending" count.
-    let mut package_qualified_imported_modules = MutSet::default();
-    for (pq_module_name, module_id) in &deps_by_name {
-        match pq_module_name {
-            PackageQualified::Unqualified(_) => {
-                package_qualified_imported_modules
-                    .insert(PackageQualified::Unqualified(*module_id));
-            }
-            PackageQualified::Qualified(shorthand, _) => {
-                package_qualified_imported_modules
-                    .insert(PackageQualified::Qualified(shorthand, *module_id));
-            }
-        }
-    }
-
-    // make sure when we run the bulitin modules in /compiler/builtins/roc that we
-    // mark these modules as Builtin. Otherwise the builtin functions are not instantiated
-    // and we just have a bunch of definitions with runtime errors in their bodies
-    let extra = {
-        match extra {
-            HeaderFor::Interface if home.is_builtin() => HeaderFor::Builtin {
-                generates_with: &[],
-            },
-            _ => extra,
-        }
-    };
-
-    (
-        home,
-        name,
-        ModuleHeader {
-            module_id: home,
-            module_path: filename,
-            is_root_module,
-            exposed_ident_ids: ident_ids,
-            module_name: loc_name.value,
-            packages: package_entries,
-            imported_modules,
-            package_qualified_imported_modules,
-            deps_by_name,
-            exposes: exposed,
-            parse_state,
-            exposed_imports: scope,
-            symbols_from_requires: Vec::new(),
-            header_for: extra,
-            module_timing,
-        },
-    )
-}
-
-#[derive(Debug)]
-struct PlatformHeaderInfo<'a> {
```

Comment:
This data has all been rolled into `HeaderInfo` - more specifically inside its `HeaderType` field (and `HeaderType` is the new name for `HeaderFor`)